src/libcore/convert.rs

```diff
@@ -48,7 +48,24 @@
 
 #![stable(feature = "rust1", since = "1.0.0")]
 
-use str::FromStr;
+use fmt;
+
+/// An uninhabited type used as the error type for implementations of fallible
+/// conversion traits in cases where they cannot actually fail.
+///
+/// Because `Infallible` has no constructors (variants), a value of this type
+/// can never exist. It is used only to satisfy trait signatures that expect
+/// an error type, and signals to both the compiler and the user that the error
+/// case is impossible.
+#[unstable(feature = "try_from", issue = "33417")]
+pub enum Infallible {}
```

Comment:
[This comment](https://github.com/rust-lang/rust/pull/42456#issuecomment-326202592) reminded me of something: Can you please add an `impl From<Infallible> for TryFromIntError`?  That way code like this will compile even if the try_into is actually infallible:
```rust
fn foo(x: c_int) -> Result<i32, TryFromIntError> { Ok(x.try_into()?) }
```